R/format-table.R

```diff
@@ -80,4 +80,31 @@ format_table <-
     }
     
     return(output_table)
-  }
\ No newline at end of file
+  }
+
+
+#' Check values for tables
+#' 
+#' Check that values present in the specified columns in the input table match the passed valid_values.
+#' Outputs a warning when case sensitive has no match but case insensitive matches.
+#' 
+#' @param input_table data.frame containing input data
+#' @param column_names vector of names of the columns within input_table to check
+#' @param valid_values vector of valid values withing the columns
+check_values_for_table <- function(input_table, column_names, valid_values){
```

Comment:
I'm confused about the what this function is doing.

1. It looks like you're checking if all the columns in a table have valid values defined by the `valid_values` argument? But aren't the valid values specific to a column? For example the way you're using this function in tables, `pK` and `fK` are not valid in the `<table_name>Order` field.
2. We should also give the row number with the invalid value 
3. I don't think there's a need to check if the approriate case is used. It should be enough to check if its a valid value or not.

Reply:
If this function is not used within the format_table function, it should be moved into its own file.

Reply:
1. Not all columns in a table are checked only columns passed in column_names are checked. Inside tables.qmd only columns with the format of <table_name> are passed. All table columns that are not "Required" or "Order" need to have the pk fk header format.
2. Row numbers are relative since the input table is a subset of parts where partType is table. Therefore returning a row number would be useless since its a rownumber within the subset table.
3. If the check is only done for valid values how do I display a warning for only pk vs pK but not for NA?

Reply:
1. Ok, that makes more sense. In that case the `column_names` variable should be changed to `columns_to_check`.
2. How would the user be able to easily find the offending value(s) then?
3. Not sure I understand? If NA is not a valid value within that column there should be an error for it?

Reply:
2. The user receives the column that contains the offending value. In excel you can check for all values within a column. This warning does not check each individual row but instead it checks values used in that column. Knowing the column and the value that caused the error is enough to filter and replace values in that column. Am I incorrect in assuming that?
3. The purpose of this function is to notify the user when they have incorrect values on things they are attempting to display. For example display for a table would include its pK fK and headers but inputs and NA are not shown. Therefore, displaying an error for values the user does not care for would clutter the warning display. Instead this function focuses on checking the specified values for minor case sensitive typos like user wanting to display pK but using pk in the dictionary. There is no intention to display NA or any other values since they are not even used in the display. In this case its best to show the user warnings related only to the passed values.